cardano-cli/src/Cardano/CLI/Read.hs

```diff
@@ -617,23 +622,19 @@ instance Error CddlError where
       prettyError e
 
 readCddlTx :: FileOrPipe -> IO (Either (FileError TextEnvelopeCddlError) CddlTx)
-readCddlTx = readFileOrPipeTextEnvelopeCddlAnyOf teTypes
+readCddlTx =
+  readFileOrPipeTextEnvelopeCddlAnyOf $
+    map (`FromCDDLTx` CddlTx) txTextEnvelopeTypes
+
+txTextEnvelopeTypes :: [Text]
+txTextEnvelopeTypes =
+  [ let TextEnvelopeType d = shelleyBasedEraConstraints sbe $ textEnvelopeType (proxyToAsType (makeTxProxy sbe))
+     in T.pack d
+  | AnyShelleyBasedEra sbe <- [minBound .. maxBound]
+  ]
  where
-  teTypes =
-    [ FromCDDLTx "Witnessed Tx ShelleyEra" CddlTx
```

Comment:
So now the types become:
```
["TxSignedShelley"
,"Tx AllegraEra"
,"Tx MaryEra"
,"Tx AlonzoEra"
,"Tx BabbageEra"
,"Tx ConwayEra"]
```
If I get this right, we won't be able to read transactions created in the previous version of cardano-cli.
I understand that it is expected?

Reply:
Yes, that is right